examples/neuroevolution-steering/vehicle.js

```diff
@@ -147,11 +157,12 @@ class Vehicle {
 
     // Create inputs
     let inputs = [];
-    // This is goofy but these 4 inputs are mapped to distance from edges
-    inputs[0] = constrain(map(this.position.x, foodBuffer, 0, 0, 1), 0, 1);
-    inputs[1] = constrain(map(this.position.y, foodBuffer, 0, 0, 1), 0, 1);
-    inputs[2] = constrain(map(this.position.x, width - foodBuffer, width, 0, 1), 0, 1);
-    inputs[3] = constrain(map(this.position.y, height - foodBuffer, height, 0, 1), 0, 1);
+    // These inputs are the location of the vehicle
+    inputs[0] = this.position.x / width;
+    inputs[1] = this.position.y / height;
+    // These inputs are the distance of the vehicle to east- and west borders
+    inputs[2] = 1 - inputs[0];
+    inputs[3] = 1 - inputs[1];
```

Comment:
Isn't this just equivalent to the AI learning negative weights/biases from `inputs[0]` and `inputs[1]`, though?

Reply:
Because all input-nodes are linked directly to all hidden-nodes I would say no in this case. If the hidden-layer was a multi-dimensional array I would say yes. The main reason for this change was because it felt a bit too 'cheaty' for me I guess and the vehicles also showed some strange behavior because the input would always be 0 until they get within this border. To be clear though I am no expert in neuro evolution, in fact I learned most of what I know about this topic from watching The Coding Train :)

Reply:
Actually I would say no in both cases. I do however agree that duplicating inputs 0 and 1 directly into 2 and 3 would have the exact same effect, but what it does do is solve the XOR-problem for the border.

Reply:
But duplicating them exactly would have no net effect. Simplified, a NN with two equivalent inputs and one middle layer node would have two weights w_11, w_21 for example, and two biases b_1, b_2, right? If so, then couldn't you build a 1-input NN with the same behaviour by just summing the weights and biases together?

I also have no idea about these things either, which is why I asked if it was actually different!

Reply:
Alright nice, so my reasoning was that with having 1 input, the NN must use a single-value to represent 3 outcomes, namely 0 and 1 for 'close to border' and 0.5 for 'not close to border'. By using 2 inputs, it can use one value to represent 2 outcomes and the other value to represent 2 outcomes. This however is wrong, as can be seen here: http://www.iciclesoft.com/preview/nn-test

The code for these testcases can be found at https://github.com/iciclesoft/NN-Test

The remaining question however is if we want to go back to just the x and y positions (like it was before) or if we want to go with the border stroke. My vote would be for the x and y positions, which they seem to pick up quite nicely after having it run for +/- one minute at 100x speed.